server/src/test/java/org/elasticsearch/gateway/DanglingIndicesStateTests.java

```diff
@@ -124,7 +154,7 @@ public void testDanglingProcessing() throws Exception {
             // check that several runs when not in the metadata still keep the dangled index around
             int numberOfChecks = randomIntBetween(1, 10);
             for (int i = 0; i < numberOfChecks; i++) {
-                Map<Index, IndexMetadata> newDanglingIndices = danglingState.findNewDanglingIndices(metadata);
+                Map<Index, IndexMetadata> newDanglingIndices = danglingState.findNewDanglingIndices(emptyMap(), metadata);
```

Comment:
I don't think this is what happens when auto-import is enabled as it is here; instead we update the `danglingIndices` field and pass that in for subsequent calls.

Reply:
Re: this and the other comment on this file...this is a weird test. The `findNewDanglingIndices` method doesn't modify anything, and yet it's being called in a loop. It makes more sense to call `findNewAndAddDanglingIndices` in a loop (which the test also does). So I feel like maybe the test need rewritten? What do you think?

Reply:
Good point, the calls to `findNewAndAddDanglingIndices` do what I suggest too. I'm ok with this as it is.